_blog/deploying-laravel-to-kubernetes/index.md

```diff
@@ -0,0 +1,289 @@
+
+RUN php artisan key:generate
+RUN php artisan serve --host=0.0.0.0 --port=8181
+EXPOSE 8181
+``` 
+This `DockerFile` is reasonably basic:
+
+It extends a PHP7 base image, installs some system dependencies including composer and the standard PHP extensions required by Laravel. It then copies the application files to a working directory and installs the application's dependencies via composer. It also runs a couple of `artisan` commands,  last of which serves the application using the built-in PHP web server. In the end, it exposes port `8181` to the host machine.
+  
+To build the local Docker image:
+
+```bash
+cd /to/your/project/directory
+eval $(minikube docker-env)
+docker build -t yourname/laravel-kubernetes-demo .
+```
+
+__Deploying the image__
+
+Now that the application's image is built and available locally you can go ahead with deploying it.
+
+I always start with making sure that Kubectl is in the correct context. In this case, the context is Minikube. You can quickly switch context as follows:
+
+```bash
+kubectl config use-context minikube
+```
+
+then you can deploy the container image:
+
+```bash
+kubectl run laravel-kubernetes-demo --image=yourname/laravel-kubernetes-demo 
+--port=8181 --image-pull-policy=IfNotPresent
+```
+The above command tells Kubectl to run our demo application from the Docker image while making port 8181 available for listening. The last parameter of the command simply asks Kubectl to not pull the image from a registry such as Docker Hub if it exists locally which in this case it does. Do note that you still need to be logged on to Docker's so that Kubectl can check if the image is up to date.
+
+You can check that a Pod is created for our application by running:
+
+```bash
+kubectl get pods
+```
+
+which should return a similar output to:
+
+```bash
+NAME                                       READY     STATUS    RESTARTS   AGE
+laravel-kubernetes-demo-7dbb9d6b48-q54wp   1/1       Running   0          18m
+```
+
+You can also use the Minikube GUI dashboard to monitor the cluster. The GUI also helps with visualising most of the discussed concepts. To view the dashboard, just run the following:
+
+```bash
+minikube dashboard
+```
+
+or to acquire the dashboard's URL address:
+
+```bash
+minikube dashboard --url=true
+```
+
+__Exposing a service__
+
+So far you have created a deployment which is running the application's container. A pod running in the cluster has a dynamic IP. If you route the traffic directly to it using the IP, you may still need to update the routing table every time the pod is restarted. In fact, on every deployment or container restart, a new IP is assigned to the pod. To avoid managing IP addresses manually, you need to use a Service. The service acts as an IP and serves as a load balancer for a set of pods. So even if the IP address of a pod changes, the service is always pointing to it. And since the Service always has the same IP, you won't need to update anything manually.
```

Comment:
The service acts as an IP and serves as a load balancer for a set of pods.

I think we could have:

The service acts as a load balancer for a set of pods.

Reply:
👍